lib/index.js

```diff
@@ -109,96 +104,104 @@ internals.Connection.prototype.validateSegmentName = function (name) {
 };
 
 
-internals.Connection.prototype.get = function (key, callback) {
-
-    if (!this.client) {
-        return callback(new Error('Connection not started'));
-    }
-
-    this.client.get(this.generateKey(key), (err, result) => {
-
-        if (err) {
-            return callback(err);
+internals.Connection.prototype.get = function (key) {
```

Comment:
You should probably make this function async, this way you can just throw the 1st error, and use the promises that ioredis already exposes. Same goes for all the functions below.